joern-cli/frontends/rubysrc2cpg/src/main/scala/io/joern/rubysrc2cpg/datastructures/RubyProgramSummary.scala

```diff
@@ -29,11 +30,12 @@ case class RubyMethod(
   returnType: String,
   baseTypeFullName: Option[String]
 ) extends MethodLike
+    derives ReadWriter
```

Comment:
Something like `baseTypeFullName` just represents the surrounding `RubyType`, so perhaps we could create a more defined serialization/serialization for `RubyMethod`? Similarly, `returnType` and `parameterTypes` can be excluded.